docs/sdk/src/guides/your_first_node.rs

```diff
@@ -1 +1,311 @@
+//! polkadot-omni-node \
+//! 	--tmp \
+//! 	--dev-block-time 100 \
+//! 	--chain <chain_spec_file>.json
+//! ```
+//!
+//! > Note that we always prefer to use `--tmp` for testing, as it will save the chain state to a
+//! > temporary folder, allowing the chain-to be easily restarted without `purge-chain`. TODO: link
+//! > to flags.
+//!
+//! This will start the node and import the blocks. Note whole using `--dev-block-time`, the node
+//! will use the testing-specific manual-seal consensus. This is an efficient way to test the
+//! application logic of your runtime, without needing to yet care about consensus, block
+//! production, relay-chain and so on.
+//!
+//! ### Next Steps
+//!
+//! * See the rest of the steps in [`crate::reference_docs::omni_node#user-journey`].
+//!
+//! [`runtime`]: crate::reference_docs::glossary#runtime
+//! [`node`]: crate::reference_docs::glossary#node
+//! [`build_config`]: first_runtime::Runtime#method.build_config
+//! [`omni-node`]: crate::reference_docs::omni_node
+//! [`--dev-block-time`]: (polkadot_omni_node_lib::cli::Cli::dev_block_time)
+
+#[cfg(test)]
+mod tests {
+	use assert_cmd::Command;
+	use rand::Rng;
+	use sc_chain_spec::{DEV_RUNTIME_PRESET, LOCAL_TESTNET_RUNTIME_PRESET};
+	use sp_genesis_builder::PresetId;
+	use std::path::PathBuf;
+
+	const PARA_RUNTIME: &'static str = "parachain-template-runtime";
+	const FIRST_RUNTIME: &'static str = "polkadot-sdk-docs-first-runtime";
+	const MINIMAL_RUNTIME: &'static str = "minimal-template-runtime";
+
+	const CHAIN_SPEC_BUILDER: &'static str = "chain-spec-builder";
+	const OMNI_NODE: &'static str = "polkadot-omni-node";
+
+	fn cargo() -> Command {
+		Command::new(std::env::var("CARGO").unwrap_or_else(|_| "cargo".to_string()))
+	}
+
+	fn get_target_directory() -> Option<PathBuf> {
+		let output = cargo().arg("metadata").arg("--format-version=1").output().ok()?;
+
+		if !output.status.success() {
+			return None;
+		}
+
+		let metadata: serde_json::Value = serde_json::from_slice(&output.stdout).ok()?;
+		let target_directory = metadata["target_directory"].as_str()?;
+
+		Some(PathBuf::from(target_directory))
+	}
+
+	fn find_release_binary(name: &str) -> Option<PathBuf> {
+		let target_dir = get_target_directory()?;
+		let release_path = target_dir.join("release").join(name);
+
+		if release_path.exists() {
+			Some(release_path)
+		} else {
+			None
+		}
+	}
+
+	fn find_wasm(runtime_name: &str) -> Option<PathBuf> {
+		let target_dir = get_target_directory()?;
+		let wasm_path = target_dir
+			.join("release")
+			.join("wbuild")
+			.join(runtime_name)
+			.join(format!("{}.wasm", runtime_name.replace('-', "_")));
+
+		if wasm_path.exists() {
+			Some(wasm_path)
+		} else {
+			None
+		}
+	}
+
+	fn maybe_build_runtimes() {
+		if find_wasm(&PARA_RUNTIME).is_none() {
+			println!("Building parachain-template-runtime...");
+			Command::new("cargo")
+				.arg("build")
+				.arg("--release")
+				.arg("-p")
+				.arg(PARA_RUNTIME)
+				.assert()
+				.success();
```

Comment:
hm, have you explored this crate:
https://github.com/krdln/command-macros

Was thinking about writing sth similar (and as usual it turned out to already be written by someone else).

We could just have:
```
command!(
  cargo build --release -p PARA_RUNTIME
).status().unwrap();
```
which is a bit easier to follow.

Reply:
This is a great suggestion, I will make a follow-up issue for it.

Reply:
https://github.com/paritytech/polkadot-sdk/issues/6194